src/Education/Commands/RuleDescriptionToolWindow.cs

```diff
@@ -37,10 +39,12 @@ internal class RuleDescriptionToolWindow : ToolWindowPane, IRuleDescriptionToolW
         public static readonly Guid ToolWindowId = new Guid(ToolWindowIdAsString);
         private readonly RuleDescriptionUserControl control;
 
-        public RuleDescriptionToolWindow()
+        public RuleDescriptionToolWindow(IServiceProvider serviceProvider)
         {
             Caption = Resources.RuleDescriptionToolWindowCaption;
-            control = new RuleDescriptionUserControl();
+            var componentModel = serviceProvider.GetService(typeof(SComponentModel)) as IComponentModel;
```

Comment:
I dont like this very much but I am thinking thats the only way I can do it

Reply:
Yep; we basically have two options:
1) this one i.e. the tool window is responsible for fetching the required service, or
2) we make the package responsible for fetching the service and passing it to the tool window when it creates it.

I would probably have gone for option (2) since the package is a "controller"/manager that is responsible for instantiating the tool window, so it doesn't feel too odd to make it responsible for supplying the tool window with the objects it needs.
It keeps the tool window simpler, and means the code for finding and creating objects is not spread out through multiple classes - it's concentrated in the packages.

However, there isn't a much in it.